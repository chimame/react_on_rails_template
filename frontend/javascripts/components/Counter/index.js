import React, { Component } from 'react'

export default class Counter extends Component {
  componentDidMount() {
    //SSRではここは呼ばれない（クライアント側では呼ばれる）なので実質2回呼ばれる
    console.log('componentDidMount')
  }

  componentWillMount() {
    //SSRでもここは呼ばれる
    console.log('componentWillMount')
  }

  render() {
    const { count, onIncrement, onDecrement } = this.props
    return (
      <div>
        <p>カウント: {count}回</p>
        <button onClick={onIncrement}>++</button>
        <button onClick={onDecrement}>--</button>
      </div>
    )
  }
}

/*
//ライフサイクルメソッド今回の例では「componentDidMount」や「componentWillMount」が必要ない場合は
//以下のように簡略してかける。
const Counter = (props) => {
  return(
    <div>
      <p>カウント: {props.count}回</p>
      <button onClick={props.onIncrement}>++</button>
      <button onClick={props.onDecrement}>--</button>
    </div>
  )
}

export default Counter
*/
