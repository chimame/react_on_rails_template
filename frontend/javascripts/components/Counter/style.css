.large {
  width: 150px;
  height: 50px;
}