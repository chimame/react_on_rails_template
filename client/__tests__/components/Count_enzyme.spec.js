import React from 'react'
import {shallow} from 'enzyme'
import { shallowToJson } from 'enzyme-to-json'
import Counter from '../../app/bundles/Sample/components/Counter/'

describe('Counter', () => {
  const subject = (incrementFn, decrementFn) => {
    return shallow(
      <Counter count={1} onIncrement={incrementFn} onDecrement={decrementFn}/>
    )
  }

  it('should be render Counter', () => {
    expect(shallowToJson(subject(jest.fn(), jest.fn()))).toMatchSnapshot()
  })

  it('should be handle increment', () => {
    const incrementMock = jest.fn()
    const decrementMock = jest.fn()
    subject(incrementMock, decrementMock).find('#increment').simulate('click')
    expect(incrementMock).toHaveBeenCalled()
    expect(decrementMock).not.toHaveBeenCalled()
  })

  it('should be handle decrement', () => {
    const incrementMock = jest.fn()
    const decrementMock = jest.fn()
    subject(incrementMock, decrementMock).find('#decrement').simulate('click')
    expect(incrementMock).not.toHaveBeenCalled()
    expect(decrementMock).toHaveBeenCalled()
  })
})
