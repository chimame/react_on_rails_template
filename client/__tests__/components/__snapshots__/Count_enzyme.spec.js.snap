exports[`Counter should be render Counter 1`] = `
<div>
  <p>
    カウント: 
    1
    回
  </p>
  <button
    className="large"
    id="increment"
    onClick={[Function]}>
    ++
  </button>
  <button
    className="large"
    id="decrement"
    onClick={[Function]}>
    --
  </button>
</div>
`;
